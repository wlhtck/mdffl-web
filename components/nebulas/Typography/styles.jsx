import { black, white } from '../colors';

const marginBottom = { margin: '0 0 32px' };
const colors = { white, black };
const families = { raleway: 'Raleway', lato: 'Lato' };

/** @function fontStyles
  * Get font styles
  * @param {Object} props - The font properties.
  * @param {('left'|'right'|'center')} [props.align='left'] - css property text-align
  * @param {('black'|'white')} [props.color='black'] - css property color (text color)
  * @param {('lato'|'raleway')} [props.family='lato'] - css property color (text color)
  * @param {string} [props.size='24px'] - css property font-size
  * @param {number} [props.weight=600] - css property font-weight */

export default ({
  align = 'left', color = 'black', family = 'lato', size = '24px', weight = 600,
} = {}) => ({
  color: colors[color],
  fontFamily: `'${families[family]}', sans-serif`,
  fontWeight: weight,
  fontSize: size,
  textAlign: align,
  ...marginBottom,
});
