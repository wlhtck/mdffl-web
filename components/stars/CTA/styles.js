import { mergeAll } from 'lodash/fp';
import { primary, white } from '../../nebulas/colors';
import fontStyles from '../../nebulas/Typography/styles';

const backgroundColors = { primary, secondary: white };
const fontColors = { primary: 'white', secondary: 'primary' };

const buttonStyles = ({ type = 'primary' } = {}) => mergeAll([
  fontStyles({
    family: 'raleway',
    size: '16px',
    weight: 800,
    color: fontColors[type],
    lineHeight: 1,
  }), {
    backgroundColor: backgroundColors[type],
    display: 'inline-block',
    padding: '16px',
    border: 'none',
    borderRadius: '4px',
    cursor: 'pointer',
    textDecoration: 'none',
  },
]);

export default buttonStyles;
