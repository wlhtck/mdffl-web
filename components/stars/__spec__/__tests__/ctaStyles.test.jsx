import styles from '../../CTA/styles';

describe('components/stars/CTA/styles', () => {
  it('returns the default styles', () => {
    expect(styles()).toMatchSnapshot();
  });

  it('returns the secondary styles', () => {
    expect(styles({ type: 'secondary' })).toMatchSnapshot();
  });
});
