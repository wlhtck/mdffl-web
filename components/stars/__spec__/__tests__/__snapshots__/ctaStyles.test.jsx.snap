// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/stars/CTA/styles returns the default styles 1`] = `
Object {
  "backgroundColor": "#1C104E",
  "border": "none",
  "borderRadius": "4px",
  "color": "#FFFFFF",
  "cursor": "pointer",
  "display": "inline-block",
  "fontFamily": "'Raleway', sans-serif",
  "fontSize": "16px",
  "fontWeight": 800,
  "lineHeight": 1,
  "margin": "0 0 32px",
  "padding": "16px",
  "textAlign": undefined,
  "textDecoration": "none",
}
`;

exports[`components/stars/CTA/styles returns the secondary styles 1`] = `
Object {
  "backgroundColor": "#FFFFFF",
  "border": "none",
  "borderRadius": "4px",
  "color": "#1C104E",
  "cursor": "pointer",
  "display": "inline-block",
  "fontFamily": "'Raleway', sans-serif",
  "fontSize": "16px",
  "fontWeight": 800,
  "lineHeight": 1,
  "margin": "0 0 32px",
  "padding": "16px",
  "textAlign": undefined,
  "textDecoration": "none",
}
`;
